packages/mui-icons-material/src/utils/SvgIcon.js

```diff
@@ -0,0 +1,237 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import {
+  unstable_capitalize as capitalize,
+  unstable_composeClasses as composeClasses,
+} from '@mui/utils';
+import { styled, useThemeProps } from '@mui/system';
+import { getSvgIconUtilityClass } from './svgIconClasses';
+
+const useUtilityClasses = (ownerState) => {
+  const { color, fontSize, classes } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      color !== 'inherit' && `color${capitalize(color)}`,
+      `fontSize${capitalize(fontSize)}`,
+    ],
+  };
+
+  return composeClasses(slots, getSvgIconUtilityClass, classes);
+};
+
+const defaultMaterialDesignColors = {
+  inherit: undefined,
+  action: 'rgba(0, 0, 0, 0.54)',
+  disabled: 'rgba(0, 0, 0, 0.26)',
+  primary: '#1976d2',
+  secondary: '#9c27b0',
+  error: '#d32f2f',
+  info: '#0288d1',
+  success: '#2e7d32',
+  warning: '#ed6c02',
+};
+
+const SvgIconRoot = styled('svg', {
+  name: 'MuiSvgIcon',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+
+    return [
+      styles.root,
+      ownerState.color !== 'inherit' && styles[`color${capitalize(ownerState.color)}`],
+      styles[`fontSize${capitalize(ownerState.fontSize)}`],
+    ];
+  },
+})(({ theme, ownerState }) => ({
+  userSelect: 'none',
+  width: '1em',
+  height: '1em',
+  display: 'inline-block',
+  fill: 'currentColor',
+  flexShrink: 0,
+  transition:
+    theme.transitions?.create?.('fill', {
+      duration: theme.transitions?.duration?.shorter,
+    }) ?? 'fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
+  fontSize: {
+    inherit: 'inherit',
+    small: theme.typography?.pxToRem?.(20) || '1.25rem',
+    medium: theme.typography?.pxToRem?.(24) || '1.5rem',
+    large: theme.typography?.pxToRem?.(35) || '2.1875rem',
+  }[ownerState.fontSize],
+  // TODO v5 deprecate, v6 remove for sx
+  color:
+    (theme.vars || theme).palette?.[ownerState.color]?.main ??
+    {
+      action: (theme.vars || theme).palette?.action?.active,
+      disabled: (theme.vars || theme).palette?.action?.disabled,
+      inherit: undefined,
+    }[ownerState.color] ??
+    defaultMaterialDesignColors[ownerState.color],
+}));
+
+/**
+ *
+ * Demos:
+ *
+ * - [Icons](https://mui.com/material-ui/icons/)
+ * - [Material Icons](https://mui.com/material-ui/material-icons/)
+ *
+ * API:
+ *
+ * - [SvgIcon API](https://mui.com/material-ui/api/svg-icon/)
+ */
+const SvgIcon = React.forwardRef(function SvgIcon(inProps, ref) {
```

Comment:
I would imagine that SvgIcon will eventually go inside `@mui/system` because once we add a new set of icons for Joy UI 1. Will we duplicate `SvgIcon` again? 2. Won't we want to have Material UI users be able to import these new icons without loading Joy UI?

Reply:
This was my first implementation too, but honestly `SvgIcon` in different libraries, will likely have different styles, this is why I've decided to move it to the icons package.

Reply:
I wouldn't put SvgIcon in the common package, at least in its current shape. It is tailored specifically for Material Design icons (sizes, viewport) and shouldn't be used with arbitrary icons (despite its generic name). If we create another icon set in the future, it can have its own SvgIcon. We could extract the common parts into a Base component if there is a lot of duplication between different packages' versions.

In general, I like the approach from #35652 a bit more. It's a change that shouldn't break anyone.

Reply:
> It is tailored specifically for Material Design icons (sizes, viewport)

I don't think that these are fundamentally specific to Material Design or Joy UI. We document how it can be used with random SVG path in https://mui.com/material-ui/icons/#font-awesome and are using it for the same use case as well. IMHO, the value of this comment is to apply a few baseline attributes and CSs that you would always want, and to provide the system style helper.

Reply:
We can publish a generic SvgIcon without the defaults tailored for Material Icons and icon set-specific versions so the sizes don't have to be overridden.

Reply:
> But then, maybe it means that we should solve https://github.com/mui/material-ui/issues/21251 and let the developers import SvgIcon from the right package?
> 
> ```
> import SvgIconMd from '@mui/material/SvgIcon';
> import SvgIconJoy from '@mui/joy/SvgIcon';
> import EditIcon from '@mui/icons-material/Edit';
> 
> <SvgIconMd>
>   <EditIcon />
> </SvgIconMd>

> ```

While this is great for a fallback method, I don't think this should be the default for a icons package we provide. I would rather have two icons packages for material & joy instead of this. However, compared to this, I think the PR's solution is still a good enough replacement.

+1 for exporting an `SvgIcon` for itegrating with third party icon sets.